torch/_functorch/aot_autograd.py

```diff
@@ -483,9 +483,15 @@ def gen_alias_from_base(aliased_base_tensor, target_meta_tensor, target_requires
     if target_meta_tensor._base is not None:
         # The base that we want to replay our view off of might have a different shape than the view's original base.
         b = target_meta_tensor._base
-        reshaped_base_tensor = aliased_base_tensor.as_strided(
-            b.size(), b.stride(), b.storage_offset()
-        )
+        abt = aliased_base_tensor
+        # Don't unnecessarily call as_strided if nothing changed; as_strided's
+        # backward is poorly implemented and slow
+        if abt.size() != b.size() or abt.stride() != b.stride() or abt.storage_offset() != b.storage_offset():
```

Comment:
not that it matters much, but a faster check would do in most cases: `if abt is not b or ...: